packages/angular_devkit/schematics/src/tree/host-tree.ts

```diff
@@ -299,6 +301,19 @@ export class HostTree implements Tree {
     return data.toString();
   }
 
+  readJSON(path: string): JsonValue | undefined {
+    const content = this.readText(path);
+    if (content === null) {
+      // undefined is returned for the not found case because null is a valid JSON value
+      return undefined;
+    }
+
+    return parse(content, undefined /* errors */, {
```

Comment:
Question: What is this `errors` argument? Is that some kind of list that `jsonc-parser` appends to with any errors? Should we validate that no errors were thrown before returning? The function declaration suggests that it will be fault tolerant and return something even when errors are thrown, which seems like implementation-specific behavior we probably don't want to expose publicly.

```typescript
/**
 * Parses the given text and returns the object the JSON content represents. On invalid input, the parser tries to be as fault tolerant as possible, but still return a result.
 * Therefore always check the errors list to find out if the input was valid.
 */
export declare function parse(text: string, errors?: {error: ParseErrorCode;}[], options?: ParseOptions): any;
```

Reply:
Something I was considering as well.
Added a check for the errors array that will throw with a message containing the error information.